src/main/java/ru/yandex/practicum/filmorate/service/interfaces/UserService.java

```
@@ -0,0 +1,17 @@
+package ru.yandex.practicum.filmorate.service.interfaces;
+
+import java.util.List;
+import ru.yandex.practicum.filmorate.models.User;
+
+// Для полноты связи перенес сюда и добавление/удаление. ПО мне - валидация должна жить в сервере, а
+// не в контроллере
```

И у контроллера и у сервиса есть некоторый интерфейс. Интерфейс определяет контракт. Контракт включает в себя допустимые значения. Поэтому, я бы сказал что зависит от того что мы валидируем. 

Если мы проверяем корректность формата входных данных, допустимый диапазон и т.д., то это должно происходить в контроллере. 

Если же мы проверяем соответствие каким-либо правилам связанным с бизнес-логикой, то в сервисе.

Не стоит забывать и про Fail Fast. Если можно проверить что-то на этапе контроллера, то незачем тянуть это дальше. Особенно учитывая что в реальных приложениях нам часто приходится разделять DTO (с ними работает контроллер) и доменные сущности (с ними работает сервис) и производить конвертацию одного в другое не проверив корректен ли запрос просто не эффективно.